### PR TITLE
xmlsec-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 687a09576b7e848b2aeb6c1e4b18e1c815bbdc4c
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T04.25-687a095/job.log
 (1/2) xmlsec-tests.py:XMLSec.test;run-run_type-upstream-68c4: ERROR: 'make' command failed. (14.70 s)
 (2/2) xmlsec-tests.py:XMLSec.test;run-run_type-distro-b4b6: FAIL: xmlsec-tests.py: 'make check' failed. (50.42 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T04.25-687a095/results.html
JOB TIME   : 76.20 s



SLES:
JOB ID     : bad0d92a30ab1b5e3f560c1b2f69863a24d13195
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T14.52-bad0d92/job.log
 (1/2) xmlsec-tests.py:XMLSec.test;run-run_type-upstream-68c4: ERROR: 'make' command failed. (13.01 s)
 (2/2) xmlsec-tests.py:XMLSec.test;run-run_type-distro-b4b6: PASS (25.63 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T14.52-bad0d92/results.html
JOB TIME   : 57.58 s
```

File: security/xmlsec-tests.py [hunk 1]
```diff
@@ -43,21 +43,34 @@ def setUp(self):
...
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = "https://github.com/lsh123/xmlsec/archive/master.zip"
+            url = self.params.get('url', default=def_url)
+            tarball = self.fetch_asset(url, expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.srcdir = os.path.join(self.workdir, 'xmlsec-master')
+            os.chdir(self.srcdir)
+            process.run('./autogen.sh', ignore_status=True)
+            if build.make(self.srcdir, process_kwargs={"ignore_status": True}):
```
Comment (abdhaleegit) [inline security/xmlsec-tests.py:55]: Just wanted to understand why ignore_status is True

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: security/xmlsec-tests.py (hunk 1 above)
Comment (nasastry) [reply security/xmlsec-tests.py:55]: @abdhaleegit
To handle 'make' error gracefully.
```
if build.make(self.srcdir, process_kwargs={"ignore_status": True}):
    self.fail("'make' command failed.")
```
If not the error from build.make will look very odd.

Comment (abdhaleegit) [reply security/xmlsec-tests.py:55]: Ok

**Review by abdhaleegit — APPROVED** [review]

**Pull request merged**